### PR TITLE
Add conda support for GQCP

### PR DESCRIPTION
**Pull request by guacke**
**Short description**
This PR will enable conda builds of GQCP.

**Related issues**
#524 #528 

**Additional context**
This Conda build will ease installations. The process of building conda packages should be brought under CD.

**Don't forget:** 
- [x] if new files are created: update the Doxygen file, the collective gqcp.hpp header and the CMake files
- [x] update the documentation on [our website](https://github.com/GQCG/gqcg.github.io) and check if new Python bindings should be added

### CODE REVIEW

**Comment by guacke** [comment]
@xdvriend @lelemmen I have uploaded the built package to Anaconda. @xdvriend While we resolve issue #528 , could you try to install GQCP in the following manner and report the output of the last command?

```bash
      conda create -c gqcg -c intel -c conda-forge -n test gqcp
      conda activate test
      python -c "import gqcpy"
```

File: conda/build.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        -DCMAKE_VERBOSE_MAKEFILE=TRUE \
+        -DBUILD_TESTS=TRUE \
+        -DBUILD_TESTS=TRUE \
+        -DBUILD_PYTHON_BINDINGS=TRUE \
+        -DPYTHON_EXECUTABLE=${PYTHON} \
+        -DPYTHON_LIBRARY=${STDLIB_DIR}/libpython${PY_VER}.a
+fi
+
+make -j${CPU_COUNT}
+make test ARGS=-j${CPU_COUNT}
```
Comment (lelemmen) [inline conda/build.sh]:
```suggestion
make test
```

I would refrain from starting multiple tests at once.

File: conda/build.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+make install
```
Comment (lelemmen) [inline conda/build.sh]: Shouldn't be only `make install` if `make test` succeeds?

File: conda_build_config.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+cxx_compiler:
+  - clangxx                       # [osx]
+  - gxx                           # [linux]
+fortran_compiler:
+  - gfortran
+target_platform:
+  - osx-64                        # [osx]
+  - linux-64                      # [linux]
+CONDA_BUILD_SYSROOT:
+  - /opt/MacOSX10.9.sdk      # [osx]
```
Comment (lelemmen) [inline conda_build_config.yaml]: Match alignment of other comments.

**Comment by xdvriend** [comment]
@guacke This is my output after following the proposed steps.
`gqcpy version 0.2.0 @ norm-1095-g5fd330d1e82287685a55b9f73b4a7b7b9eae1d96`

**Comment by xdvriend** [comment]
@guacke When trying to use gqcpy after installing it this way, i get an unknown exception. @lelemmen told me this is because something went wrong in the compilation of gqcpy. 
I included a screenshot with all the info of my commands and where the error occurred. I'm working in a notebook that I made in my virtual environment where gqcpy was installed. 

Screenshot 2020-02-20 at 15 58 09

**Pull request merged**